src/participant.rs

```diff
@@ -286,28 +286,32 @@ pub(crate) trait InnerProtocolParticipant: ProtocolParticipant {
             .collect()
     }
 
-    /// Process a `ready` message: tell other participants that we're ready and
-    /// see if all others have also reported that they are ready.
+    /// Process a `ready` message: If it came from another Participant, ignore
+    /// it. Once ready, process all messages that were received before
+    /// ready.
```

Comment:
nit: It's more of a "rejection" than an "ignore"; this would imply another party is actively misbehaving (even if it's not in a particularly useful way) and by returning a protocol error, we're not passively letting it happen.